src/compiler/factory.ts

```diff
@@ -75,6 +75,11 @@ namespace ts {
         if (typeof value === "number") {
             return createNumericLiteral(value + "");
         }
+        if (typeof value as string === "bigint") {

```

Comment:
we need to remove the cast, can you create a list in the issue to track changes after LKG updates.

Reply:
I do not think we should be checking the typeof the value. we should be able to compile programs with `bigint` even if running on older engines. 

I think you should revert all `typeof ... = "bigint"` changes in the code.